src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -404,5 +408,26 @@ private static void OnRequestError(WinHttpRequestState state, Interop.WinHttp.WI
                     break;
             }
         }
+
+        private static void ResetAuthRequestHeaders(WinHttpRequestState state)
+        {
+            const string AuthHeaderNameWithColon = "Authorization:";
+            SafeWinHttpHandle requestHandle = state.RequestHandle;
+            state.RequestMessage.Headers.Authorization = null;
```

Comment:
Mutating the original RequestMessage like this isn't usually done.  You've already removed the Auth header at the WinHTTP layer.

Reply:
That makes sense. I'll make the change.